src/CattyTests/Bricks/SetPenColorBrickTests.swift

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+final class SetPenColorBrickTests: AbstractBrickTest {
+
+    func testSetPenColorBrick() {
```

Comment:
Please add additional tests verifying what is happening when this Brick is executed with negative values, string formulas, values greater than INT_MAX, or smaller than INT_MIN (e.g., `testSetPenColorBrickWithFormulaLargerThanIntMax`, `testSetPenColorBrickWithNegativeFormula`, `testSetPenColorBrickWithStringFormula`)

Reply:
In the UIColor extension, the convenience init was previously checking whether the values are in range using assert statements. So to make it consistent with the behaviour on Catroid, I have removed those statements, as UIColor handles it internally and it is consistent with Catroid as well. That also includes negative values. For string values, the formula editor will give a syntax error so that isn't a possibility. Please let me know if I am missing out anything.

Reply:
You are using `interpretInteger` in your implementation which is very clever. Other Bricks do not do that and manually cast the result (double) to integer which of course  is crashing the app for out-of-range values. That was one of the reason we needed to make a hotfix release last week. So please verify that your Brick also works with those two out-of-range values (greater `INT_MAX` and smaller `INT_MIN`) -- it is a necessary documentation in case someone removes or refactors the `interpretInteger` or your Brick instruction.

I have a feeling that the app crashes with your Brick having a Formula evaluating to smaller than `INT_MIN`. Thanks a lot for pointing that out -- that probably just saved us a lot of time debugging that bug (ideally it would be awesome if you can also add `testInterpretIntegerMin` to `FormulaManagerInterpreterTests` then you know what I am talking about).